public/javascripts/signup-phone.js

```diff
@@ -0,0 +1,117 @@
+(function() {
+  function InputField($inputField, regex, type) {
+    const cancelable = $inputField.classList.contains('cancelable');
+  
+    const $input = $inputField.querySelector('input');
+    const $cancelButton = cancelable ? $inputField.querySelector('.cancel') : null;
+  
+    const handleFieldFocusChange = () => {
+      $inputField.classList.add('focus');
+    };
+
+    this.disable = () => {
+      $input.disabled = 'disabled'
+    }
+
+    this.validate = () => {
+      if (regex && regex.constructor === RegExp) {
+        if (regex.test($input.value)) {
+          $inputField.classList.remove('invalid')
+          $inputField.classList.add('valid')
+          if (this.onvalid && this.onvalid.constructor == Function) {
+            this.onvalid();
+          }
+          return true;
+        } else {
+          $inputField.classList.remove('valid')
+          $inputField.classList.add('invalid')
+          if (this.oninvalid && this.oninvalid.constructor == Function) {
+            this.oninvalid();
+          }
+          return false;
+        }
```

Comment:
```suggestion
        if (!regex.test($input.value)) {
          $inputField.classList.remove('valid')
          $inputField.classList.add('invalid')
          if (this.oninvalid && this.oninvalid.constructor == Function) {
            this.oninvalid();
          }
          return false;
        }
        
        $inputField.classList.remove('invalid')
        $inputField.classList.add('valid')
        if (this.onvalid && this.onvalid.constructor == Function) {
          this.onvalid();
        }
        return true;
```

가독성 및 `if-else` 중첩 `depth`를 낮추기 위해

위와 같은 구조로 변경하는 것이 어떨까요?